### PR TITLE
Update to C# 7

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs [hunk 1]
```diff
@@ -96,11 +95,7 @@ protected static MatchResult HandleDeclaration(SyntaxNodeAnalysisContext context
                 var summaryNodes = summaryElement.Nodes().ToList();
                 if (summaryNodes.Count >= 3)
                 {
-                    var firstTextPartNode = summaryNodes[0] as XText;
-                    var classReferencePart = summaryNodes[1] as XElement;
-                    var secondTextPartNode = summaryNodes[2] as XText;
-
-                    if (firstTextPartNode != null && classReferencePart != null && secondTextPartNode != null)
+                    if (summaryNodes[0] is XText firstTextPartNode && summaryNodes[1] is XElement classReferencePart && summaryNodes[2] is XText secondTextPartNode)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs]: A couple more parenthesis will improve readability quite a bit here.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs [hunk 2]
```diff
@@ -128,11 +122,7 @@ protected static MatchResult HandleDeclaration(SyntaxNodeAnalysisContext context
                 if (summaryElement.Content.Count >= 3)
                 {
                     // Standard text has the form <part1><see><part2>
-                    var firstTextPartSyntax = summaryElement.Content[0] as XmlTextSyntax;
-                    var classReferencePart = summaryElement.Content[1] as XmlEmptyElementSyntax;
-                    var secondTextPartSyntax = summaryElement.Content[2] as XmlTextSyntax;
-
-                    if (firstTextPartSyntax != null && classReferencePart != null && secondTextPartSyntax != null)
+                    if (summaryElement.Content[0] is XmlTextSyntax firstTextPartSyntax && summaryElement.Content[1] is XmlEmptyElementSyntax classReferencePart && summaryElement.Content[2] is XmlTextSyntax secondTextPartSyntax)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs]: A couple more parenthesis will improve readability quite a bit here.

File: StyleCop.Analyzers/StyleCop.Analyzers/Helpers/UsingDirectiveSyntaxHelpers.cs [hunk 3]
```diff
@@ -104,9 +104,7 @@ internal static bool StartsWithAlias(this UsingDirectiveSyntax usingDirective, S
         {
             foreach (var identifier in usingDirective.DescendantNodes())
             {
-                IdentifierNameSyntax identifierName = identifier as IdentifierNameSyntax;
-
-                if (identifierName != null && ExcludeGlobalKeyword(identifierName))
+                if (identifier is IdentifierNameSyntax identifierName && ExcludeGlobalKeyword(identifierName))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/Helpers/UsingDirectiveSyntaxHelpers.cs]: I would prefer to have the first term in parenthesis for readability

File: StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518UseLineEndingsCorrectlyAtEndOfFile.cs [hunk 4]
```diff
@@ -121,8 +121,7 @@ private static void HandleSyntaxTree(SyntaxTreeAnalysisContext context, StyleCop
 
             if (precedingTrivia.IsDirective)
             {
-                DirectiveTriviaSyntax directiveTriviaSyntax = precedingTrivia.GetStructure() as DirectiveTriviaSyntax;
-                if (directiveTriviaSyntax != null && directiveTriviaSyntax.EndOfDirectiveToken.HasTrailingTrivia)
+                if (precedingTrivia.GetStructure() is DirectiveTriviaSyntax directiveTriviaSyntax && directiveTriviaSyntax.EndOfDirectiveToken.HasTrailingTrivia)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518UseLineEndingsCorrectlyAtEndOfFile.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs [hunk 5]
```diff
@@ -154,15 +153,13 @@ private static void HandleParenthesizedExpression(SyntaxNodeAnalysisContext cont
                     }
                     else
                     {
-                        EqualsValueClauseSyntax equalsValue = node.Parent as EqualsValueClauseSyntax;
-                        if (equalsValue != null && equalsValue.Value == node)
+                        if (node.Parent is EqualsValueClauseSyntax equalsValue && equalsValue.Value == node)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1119StatementMustNotUseUnnecessaryParenthesis.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1101PrefixLocalCallsWithThis.cs [hunk 6]
```diff
@@ -177,8 +177,7 @@ private static void HandleIdentifierNameImpl(SyntaxNodeAnalysisContext context,
                     return;
                 }
 
-                IMethodSymbol methodSymbol = symbol as IMethodSymbol;
-                if (methodSymbol != null && methodSymbol.MethodKind == MethodKind.Constructor)
+                if (symbol is IMethodSymbol methodSymbol && methodSymbol.MethodKind == MethodKind.Constructor)
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1101PrefixLocalCallsWithThis.cs]: A couple more parenthesis would improve readability

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1122UseStringEmptyForEmptyStrings.cs [hunk 7]
```diff
@@ -102,14 +100,12 @@ private static bool HasToBeConstant(LiteralExpressionSyntax literalExpression)
                     return false;
                 }
 
-                FieldDeclarationSyntax fieldDeclarationSyntax = variableDeclarationSyntax.Parent as FieldDeclarationSyntax;
-                if (fieldDeclarationSyntax != null && fieldDeclarationSyntax.Modifiers.Any(SyntaxKind.ConstKeyword))
+                if (variableDeclarationSyntax.Parent is FieldDeclarationSyntax fieldDeclarationSyntax && fieldDeclarationSyntax.Modifiers.Any(SyntaxKind.ConstKeyword))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1122UseStringEmptyForEmptyStrings.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1122UseStringEmptyForEmptyStrings.cs [hunk 8]
```diff
@@ -102,14 +100,12 @@ private static bool HasToBeConstant(LiteralExpressionSyntax literalExpression)
...
                 {
                     return true;
                 }
 
-                LocalDeclarationStatementSyntax localDeclarationStatementSyntax = variableDeclarationSyntax.Parent as LocalDeclarationStatementSyntax;
-                if (localDeclarationStatementSyntax != null && localDeclarationStatementSyntax.Modifiers.Any(SyntaxKind.ConstKeyword))
+                if (variableDeclarationSyntax.Parent is LocalDeclarationStatementSyntax localDeclarationStatementSyntax && localDeclarationStatementSyntax.Modifiers.Any(SyntaxKind.ConstKeyword))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/ReadabilityRules/SA1122UseStringEmptyForEmptyStrings.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/DocumentationRules/SA1617CodeFixProvider.cs [hunk 9]
```diff
@@ -83,8 +83,7 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
             List<SyntaxNode> nodesToFix = new List<SyntaxNode>();
             nodesToFix.Add(returnsElement);
 
-            var previousAsTextSyntax = previous as XmlTextSyntax;
-            if (previousAsTextSyntax != null && XmlCommentHelper.IsConsideredEmpty(previousAsTextSyntax))
+            if (previous is XmlTextSyntax previousAsTextSyntax && XmlCommentHelper.IsConsideredEmpty(previousAsTextSyntax))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/DocumentationRules/SA1617CodeFixProvider.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1615ElementReturnValueMustBeDocumented.cs [hunk 10]
```diff
@@ -84,9 +84,7 @@ private static void HandleDeclaration(SyntaxNodeAnalysisContext context, bool ne
                 return;
             }
 
-            var predefinedType = returnType as PredefinedTypeSyntax;
-
-            if (predefinedType != null && predefinedType.Keyword.IsKind(SyntaxKind.VoidKeyword))
+            if (returnType is PredefinedTypeSyntax predefinedType && predefinedType.Keyword.IsKind(SyntaxKind.VoidKeyword))
```
Comment (vweijsters) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1615ElementReturnValueMustBeDocumented.cs]: I would prefer to see the first term in parenthesis

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs (hunk 1 above)
Comment (sharwell) [reply StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/StandardTextDiagnosticBase.cs]: ❓ Is it sufficient to break the clauses to one clause per line, or are you specifically interested in parentheses?

**Pull request merged**